framework/src/org/checkerframework/framework/util/QualifierPolymorphism.java

```diff
@@ -392,8 +417,7 @@ protected Void scan(AnnotatedTypeMirror type, Void p) {
 
             assert actualType.getKind() == type.getKind();
             type = (AnnotatedDeclaredType) AnnotatedTypes.asSuper(types, atypeFactory, type, actualType);
-            // TODO: type is null if type is an intersection type
-            // assert type != null;
+
             if (type == null) {
```

Comment:
Now that intersection types are already explicitly handled at the beginning of this method, can this be strengthened to an errorAbort if the type is null?